fdbserver/LogRouter.actor.cpp

```diff
@@ -344,6 +346,11 @@ ACTOR Future<Reference<ILogSystem::IPeekCursor>> getPeekCursorData(LogRouterData
 				TraceEvent("LogRouterPeekLocation", self->dbgid)
 				    .detail("LogID", result->getPrimaryPeekLocation())
 				    .trackLatest(self->eventCacheHolder->trackingKey);
+				// If no primary peek location after many tries, flag an error for manual intervention.
+				// The LR may become a bottleneck on the system and need to be excluded.
+				if (!self->primaryPeekLocation.present() && !(++noPrimaryPeekLocation % 4)) {
```

Comment:
It seems `noPrimaryPeekLocation` should be reset if primary location is found. This will give us the signal when this problem happens again, so it's good to have the added trace event.